src/test/scala/com/microsoft/hyperspace/index/HybridScanForNonPartitionedDataSuite.scala

```diff
@@ -99,71 +38,121 @@ class HybridScanForNonPartitionedDataSuite extends HybridScanTestSuite {
     val indexConfig = IndexConfig("index_ParquetDelete2", Seq("clicks"), Seq("query"))
     Seq(("indexWithoutLineage", "false", false), ("indexWithLineage", "true", true)) foreach {
       case (indexName, lineageColumnConfig, transformationExpected) =>
-        withSQLConf(IndexConstants.INDEX_LINEAGE_ENABLED -> lineageColumnConfig) {
-          setupIndexAndChangeData(
-            fileFormat,
-            sampleDataFormatDelete2,
-            indexConfig.copy(indexName = indexName),
-            appendCnt = 0,
-            deleteCnt = 1)
-
-          val df = spark.read.format(fileFormat).load(sampleDataFormatDelete2)
-          def filterQuery: DataFrame =
-            df.filter(df("clicks") <= 2000).select(df("query"))
-          val baseQuery = filterQuery
-          val basePlan = baseQuery.queryExecution.optimizedPlan
-          withSQLConf(
-            IndexConstants.INDEX_HYBRID_SCAN_ENABLED -> "true",
-            IndexConstants.INDEX_HYBRID_SCAN_DELETE_ENABLED -> "false") {
-            val filter = filterQuery
-            assert(basePlan.equals(filter.queryExecution.optimizedPlan))
-          }
-          withSQLConf(
-            IndexConstants.INDEX_HYBRID_SCAN_ENABLED -> "true",
-            IndexConstants.INDEX_HYBRID_SCAN_DELETE_ENABLED -> "true") {
-            val filter = filterQuery
-            assert(
-              basePlan
-                .equals(filter.queryExecution.optimizedPlan)
-                .equals(!transformationExpected))
+        withTempPathAsString { testPath =>
+          withSQLConf(IndexConstants.INDEX_LINEAGE_ENABLED -> lineageColumnConfig) {
+            setupIndexAndChangeData(
+              fileFormat,
+              testPath,
+              indexConfig.copy(indexName = indexName),
+              appendCnt = 0,
+              deleteCnt = 1)
+
+            val df = spark.read.format(fileFormat).load(testPath)
+
+            def filterQuery: DataFrame =
+              df.filter(df("clicks") <= 2000).select(df("query"))
+
+            val baseQuery = filterQuery
+            val basePlan = baseQuery.queryExecution.optimizedPlan
+            withSQLConf(
+              IndexConstants.INDEX_HYBRID_SCAN_ENABLED -> "true",
+              IndexConstants.INDEX_HYBRID_SCAN_DELETE_ENABLED -> "false") {
+              val filter = filterQuery
+              assert(basePlan.equals(filter.queryExecution.optimizedPlan))
+            }
+            withSQLConf(
+              IndexConstants.INDEX_HYBRID_SCAN_ENABLED -> "true",
+              IndexConstants.INDEX_HYBRID_SCAN_DELETE_ENABLED -> "true") {
+              val filter = filterQuery
+              assert(
+                basePlan
+                  .equals(filter.queryExecution.optimizedPlan)
+                  .equals(!transformationExpected))
+            }
           }
         }
     }
   }
+
+  test("Delete-only: filter rule, number of delete files threshold") {
```

Comment:
Note: moved from HybridScanTestSuite as no need to run this test for different source file formats.